packages/hardhat/contracts/CoinEngine.sol

```diff
@@ -93,10 +95,7 @@ contract MyUSDEngine is Ownable {
             // Update exchange rate to reflect new value
             debtExchangeRate += (interest * PRECISION) / totalDebtShares;
 
-            // Mint interest to the staking contract
-            i_myUSD.mintTo(address(i_staking), interest);
-
-            // Update the staking contract's exchange rate to reflect the new interest
+            // Update the staking contract's interest rate
             i_staking.setInterestRate(myUSDSavingsRate);
```

Comment:
It feels disjointed to set the staking contract interest rate here. I would think we should move this to `setSavingsRate`. And then I would question whether we need to run `_accrueInterest` in `setSavingsRate` since it shouldn't impact borrow interest accrued. When the savings rate is set it should trigger the corresponding `_accrueInterest` method in the staking contract.